packages/realm/src/Logger.ts

```diff
@@ -186,6 +186,33 @@ export type LogEntry = {
 export type LoggerCallback2 = (entry: LogEntry) => void;
 export type LoggerCallback = LoggerCallback1 | LoggerCallback2;
 
+/** @internal */
+export function toBindingLogger(logger: LoggerCallback) {
+  if (isLoggerWithLevel(logger)) {
+    return binding.Helpers.makeLogger((_, level, message) => {
+      logger(fromBindingLoggerLevelToLogLevel(level), message);
+    });
+  }
+  if (isLoggerWithOptions(logger)) {
+    return binding.Helpers.makeLogger((category, level, message) => {
+      logger({
+        category: category as LogCategory,
+        level: fromBindingLoggerLevelToLogLevel(level),
+        message,
+      });
+    });
+  }
+  throw new Error(`Unexpected logger passed.`);
+}
+
+function isLoggerWithLevel(logger: unknown): logger is LoggerCallback1 {
+  return typeof logger === "function" && logger.length === 2;
+}
+
+function isLoggerWithOptions(logger: unknown): logger is LoggerCallback2 {
+  return typeof logger === "function" && logger.length === 1;
+}
```

Comment:
Great use if type guards! I suppose you know `logger` will be some function, so you could make it even more tight, but either way would be fine 👍 

```suggestion
function isLoggerWithLevel(logger: Function): logger is LoggerCallback1 {
  return logger.length === 2;
}

function isLoggerWithOptions(logger: Function): logger is LoggerCallback2 {
  return logger.length === 1;
}
```

Reply:
With the current implementation we actually don't know that it's a function before those are called. But the caller of `toBindingLogger()` could do an `assert()` first (I'll add that). Then those functions can assume it's a function like you said 🙂